core/checkcloudconnection.cpp

```diff
@@ -91,6 +94,38 @@ void CheckCloudConnection::sslErrors(const QList<QSslError> &errorList)
 		qDebug() << err.errorString();
 }
 
+bool CheckCloudConnection::nextServer()
+{
+	struct serverTried {
+		const char *server;
+		bool tried;
+	};
+	static struct serverTried cloudServers[] = {
+		{ CLOUD_HOST_EU, false },
+		{ CLOUD_HOST_US, false }
+	};
+	const char *server = nullptr;
+	for (int i = 0; i < CLOUD_NUM_HOSTS; i++) {
```

Comment:
Ugh. This loop will start failing in odd ways (possible random memory corruption etc) if CLOUD_NUM_HOSTS is ever changed without changing the cloudServers[] array.

Which is really easy to do by mistake when adding a server.

So rather than have a separate #define for a number in a header file that can get out of sync, why not use ARRAY_SIZE() (currently only used by the git code in load-git.c).

Reply:
BRILLIANT. Good catch. Yes. Thanks. Will do.